godtools/App/Features/AppLanguage/Presentation/DownloadableLanguages/Subviews/DownloadableLanguageItem/DownloadableLanguageItemViewModel.swift

```diff
@@ -0,0 +1,256 @@
+    private static var resetIsMarkedForRemovalTimers: [LanguageId: SwiftUITimer] = Dictionary()
+    private static var backgroundCancellables: Set<AnyCancellable> = Set()
+    
+    private let downloadToolLanguageUseCase: DownloadToolLanguageUseCase
+    private let removeDownloadedToolLanguageUseCase: RemoveDownloadedToolLanguageUseCase
+    
+    private var cancellables: Set<AnyCancellable> = Set()
+    
+    private weak var flowDelegate: FlowDelegate?
+    
+    let downloadableLanguage: DownloadableLanguageListItemDomainModel
+    let recycleState: DownloadableLanguageItemRecycleState
+    
+    @Published private(set) var downloadState: DownloadableLanguageDownloadState
+    @Published private(set) var iconState: LanguageDownloadIconState = .notDownloaded
+    
+    init(flowDelegate: FlowDelegate, downloadableLanguage: DownloadableLanguageListItemDomainModel, downloadToolLanguageUseCase: DownloadToolLanguageUseCase, removeDownloadedToolLanguageUseCase: RemoveDownloadedToolLanguageUseCase) {
+        
+        self.flowDelegate = flowDelegate
+        self.downloadableLanguage = downloadableLanguage
+        self.downloadToolLanguageUseCase = downloadToolLanguageUseCase
+        self.removeDownloadedToolLanguageUseCase = removeDownloadedToolLanguageUseCase
+        
+        let languageId: String = downloadableLanguage.languageId
+        
+        if let existingRecycleState = Self.inMemoryStateForRecycle[languageId] {
+            recycleState = existingRecycleState
+        }
+        else {
+            recycleState = DownloadableLanguageItemRecycleState()
+            Self.inMemoryStateForRecycle[languageId] = recycleState
+        }
+        
+        downloadState = Self.getInitialDownloadState(
+            downloadableLanguage: downloadableLanguage,
+            recycleState: recycleState
+        )
+                
+        Publishers.CombineLatest(
+            $downloadState,
+            recycleState.$isMarkedForRemoval
+        )
+        .flatMap { (downloadState: DownloadableLanguageDownloadState, isMarkedForRemoval: Bool) -> AnyPublisher<LanguageDownloadIconState, Never>  in
+            
+            let iconState: LanguageDownloadIconState
+            
+            if isMarkedForRemoval {
+                iconState = .remove
+            }
+            else {
+                
+                switch downloadState {
+                case .downloaded:
+                    iconState = .downloaded
+                case .downloading(let progress):
+                    iconState = .downloading(progress: progress)
+                case .notDownloaded:
+                    iconState = .notDownloaded
+                }
+            }
+            
+            return Just(iconState)
+                .eraseToAnyPublisher()
+        }
+        .assign(to: &$iconState)
+        
+        recycleState
+            .$downloadProgress
+            .sink(receiveValue: { [weak self] (progress: Double?) in
```

Comment:
Hey @rachaelblue note that I didn't add receive on main queue here because I noticed it would cause the UI to flicker when tapping multiple languages to download because I think the ViewModel was getting re-allocated when tapping additional languages.
The static observers will call receive on main when updating the recycle state object.